subsys/bluetooth/services/ots/ots.c

```diff
@@ -495,6 +546,65 @@ BT_GATT_SERVICE_INSTANCE_DEFINE(ots_service_list, ots_instances,
 				CONFIG_BT_OTS_MAX_INST_CNT,
 				BT_GATT_OTS_ATTRS);
 
+static void ots_delete_empty_name_objects(struct bt_ots *ots)
+{
+	char id_str[BT_OTS_OBJ_ID_STR_LEN];
+	struct bt_gatt_ots_object *obj;
+	struct bt_gatt_ots_object *next_obj;
+	int err;
+
+	err = bt_gatt_ots_obj_manager_first_obj_get(ots->obj_manager, &next_obj);
+	while (!err) {
+		obj = next_obj;
+
+		/* Get the next object before we potentially delete the current object and
+		 * no longer can get the next object
+		 */
+		err = bt_gatt_ots_obj_manager_next_obj_get(ots->obj_manager, obj, &next_obj);
+
+		if (strlen(obj->metadata.name) == 0) {
+			bt_ots_obj_id_to_str(obj->id, id_str, sizeof(id_str));
+			LOG_DBG("Deleting object with %s ID due to empty name", log_strdup(id_str));
+
+			if (ots->cb->obj_deleted) {
+				ots->cb->obj_deleted(ots, NULL, obj->id);
+			}
```

Comment:
Consider passing the `conn` pointer of the connection that causes this to be called, and forward this to the upper layers in this callback.

Reply:
Oops! Thanks!